pkg/confgen/encode.go

```diff
@@ -40,17 +40,7 @@ func (cg *ConfGen) parseEncode(encode *map[string]interface{}) (*api.PromEncode,
 		log.Debugf("Unmarshal aggregate.Definitions err: %v ", err)
 		return nil, err
 	}
-
-	// validate that ValueKey is unique
-	for _, existingPromMetric := range cg.promMetrics {
-		for _, newPromMetric := range prom.Metrics {
-			if existingPromMetric.ValueKey == newPromMetric.ValueKey {
-				log.Panicf("error in parseEncode: ValueKey %s overlaps, metric encoding ignored !!!! ", newPromMetric.ValueKey)
-				return nil, err
-			}
-		}
-	}
-
+	
```

Comment:
Any other cross phases check that you think we can and should add here???

Reply:
Consider the following configuration of a prom metric
```yaml
      - name: bandwidth_per_network_service
        type: counter
        filter:
          key: name
          value: bandwidth_network_service
        valuekey: recent_op_value
        labels:
        - by
        - aggregate
        buckets: []
```


If `prom_encode` follows `extract_aggregate` I can think of the following validations:
1. `valuekey` is one of the keys of the following GenericMap https://github.com/netobserv/flowlogs-pipeline/blob/c0f17db3e2619c2cb95b6c945f2e7fa2982fed68/pkg/pipeline/extract/aggregate/aggregate.go#L189-L199
2. Same is true about `labels` and `filter.key`
3. if `filter.key == name` then we can validate that `filter.value` is defined.
4. regarding `type`
4.1. if `type == gauge` then `valuekey` is in `[total_value, total_count]`
4.2. if `type == counter` then `valuekey` is in `[recent_op_value, recent_count]`
4.3. if `type == histogram` then `valuekey` is `recentRawValues`
5. histograms have at least one bucket